Chore: frontend emit type info

Laurin-W opened the pull request:
Right now, the parcel build does not emit type information which causes problems when working with typescript and importing frontend components.

With this commit, all index file types are changed from `.js` to `.ts`. And a record `"types": "dist/index.d.ts"` is added.

This way, inferred type information is exported for `js` files as well to `dist/index.d.ts`.

Laurin-W committed:
chore: frontend: packages emit type info (d.ts)

Laurin-W committed:
chore: frontend: build

Laurin-W committed:
chore: frontend: make build script ran yalc:publish

Laurin-W committed:
chore: frontend: update eslint ignore rule

mguihal reviewed (approved):
Nice!

How will it work with other types file like https://github.com/assemblee-virtuelle/semapps/blob/next/src/frontend/%40types/semantic-data-provider.d.ts ? Which one will be used when importing this package in another here? We may remove it, but types are more specific than the generated file

Laurin-W committed:
chore: frontend: remove manual @types declarations

Laurin-W commented:
> How will it work with other types file like https://github.com/assemblee-virtuelle/semapps/blob/next/src/frontend/%40types/semantic-data-provider.d.ts ? Which one will be used when importing this package in another here? We may remove it, but types are more specific than the generated file

Thanks for pinpointing, I didn't know of that file.
It seems, that this is not regarded in the export. I removed your referenced declaration file and changed the file to typescript instead and added the necessary type info, this way it will be kept up to date.

Doing that, I noticed, there are some eslint rules to fix (importing js files from ts gives warnings) but that's for another time.

From now on, it's probably the easiest to change a file to `.ts` and add the type information where needed :)
